File: programmers/lv2/귤_고르기.py
def solution(k, tangerine):
    answer = 0
    t_dict = {}
    for t in tangerine:
        if t in t_dict:
            t_dict[t] += 1
        else:
            t_dict[t] = 1
    sorted_t = sorted(t_dict.items(), key=lambda x: x[1], reverse=True)
    cnt = 0
    for key, value in sorted_t:
        cnt += value
        answer += 1
        if k <= cnt:
            return answer

File: programmers/lv2/귤_고르기.js
function solution(k, tangerine) {
  let answer = 0;
  const size = {};
  for (const t of tangerine) {
    if (t in size) size[t] += 1;
    else size[t] = 1;
  }
  const sizeArr = Object.values(size);
  sizeArr.sort((a, b) => b - a);
  for (let i = 0; i < sizeArr.length; i++) {
    k -= sizeArr[i];
    answer += 1;
    if (k <= 0) break;
  }
  return answer;
}
